third_party/move/tools/move-resource-viewer/src/lib.rs

```diff
@@ -246,16 +247,23 @@ impl<V: CompiledModuleView> MoveValueAnnotator<V> {
         &self,
         tag: &StructTag,
         blob: &[u8],
-    ) -> anyhow::Result<Vec<(Identifier, MoveValue)>> {
+    ) -> anyhow::Result<(Option<Identifier>, Vec<(Identifier, MoveValue)>)> {
         let ty = self.resolve_struct(tag)?;
         let struct_def = (&ty).try_into().map_err(into_vm_status)?;
         Ok(match MoveStruct::simple_deserialize(blob, &struct_def)? {
-            MoveStruct::Runtime(runtime) => self
-                .get_field_names(&ty)?
-                .into_iter()
-                .zip(runtime)
-                .collect(),
-            MoveStruct::WithFields(fields) | MoveStruct::WithTypes { fields, .. } => fields,
+            MoveStruct::Runtime(values) => {
+                let (_, field_names) = self.get_field_information(&ty, None)?;
```

Comment:
I added a debug assert, but this is clearly satisfied invariant since if you pass in None to this function, the returned tag will always be None. Does not depend on external input.